pkg/v1/tkg/test/tkgctl/shared/e2e_common.go

```diff
@@ -223,3 +269,69 @@ func E2ECommonSpec(ctx context.Context, inputGetter func() E2ECommonSpecInput) {
 		By("Test successful !")
 	})
 }
+
+func getNextAvailableTkrVersion(ctx context.Context, currentTkrVersion string, mngClient client.Client) (string, error) {
+	tkrList := runv1alpha3.TanzuKubernetesReleaseList{}
+	mngClient.List(ctx, &tkrList)
+	tkrs := tkrList.Items
+	availableTkrs := make(map[string]string)
+	for i := range tkrs {
+		compatible := ""
+		for _, condition := range tkrs[i].Status.Conditions {
+			if condition.Type == runv1alpha3.ConditionCompatible {
+				compatible = string(condition.Status)
```

Comment:
Should you break from the for loop if compatible was found?

Reply:
nope. we need to get a list of all compatibles.  not just one, because we have no way of ensuring the list is in order. So we make a list of all compatibles, and then we pick the next one from the current tkr.

Reply:
@adduarte I believe this comment still applies. ```compatible``` is just a string and not an array anyway, you can just break from the ```for _, condition := range tkrs[i].Status.Conditions ``` when found it

Reply:
followed suggestions